app/src/main/res/values/strings.xml

```diff
@@ -699,5 +699,6 @@
     <string name="legacy_listing_summary">If enabled, uses stat for listing, else uses ls</string>
     <string name="error_unsupported_v5_rar">RAR archive \"%s\" is unsupported RAR v5 archive.</string>
     <string name="ftp_prompt_restart_server">You will need to restart the FTP server for changes to take effect.</string>
+    <string name="smb_share">Share path (optional)</string>
```

Comment:
Wouldn't it be better to use string formatting and declare `optional` as a separate text?
`<string name="smb_share">Share path (%s)</string>
<string name="optional">optional</string>
`

Reply:
@VishalNehra  I worry this may not be as useful, because trying to reuse the domain and smb_share strings without them being optional (perhaps in the future) will result in an empty set of parenthesis. Perhaps something like this is better:

strings.xml
```
<string name="smb_share">Share path %s</string>
<string name="optional">(optional)</string>
```

smb_dialog.xml
```
android:hint="@{@string/smb_share(@string/optional)}"
```

(Please let me know if my syntax or understanding is off here. I am not an android dev and this is my first time using this type of string formatting in java xml files.... just trying to create the most extensible/reusable code here)

Reply:
Also want to point out [this line](https://github.com/TeamAmaze/AmazeFileManager/blob/f7899f138504ef7a956da59a66a7a6b2d87932b5/app/src/main/java/com/amaze/filemanager/ui/dialogs/SmbConnectDialog.java#L131) which I *think* will cause a regression in the exact way I described (empty parenthesis)

Reply:
Agreed. In that case I think appending an asterisk on code side (not xml side) in front of mandatory fields would be better?